dsa-solutions/Sorting-Algorithms/Counting-Sort.md

```diff
@@ -0,0 +1,122 @@
+---
+id: Counting-Sort
+title: Counting Sort (Geeks for Geeks)
+sidebar_label: Counting Sort
+tags:
+  - Beginner
+  - Sorting Algorithms
+  - Geeks for Geeks
+  - CPP
+  - Python
+  - Java
+  - JavaScript
+  - DSA
+description: "This is a solution to the Counting Sort problem on Geeks for Geeks."
+---
+
+## 1. What is Counting Sort?
+
+Counting Sort is a non-comparison-based sorting algorithm. It works by counting the number of occurrences of each unique element in the input array. The algorithm uses this count information to place the elements in their correct positions in the output array.
+
+## 2. Algorithm for Counting Sort
+
+1. Find the maximum element in the input array.
+2. Create a count array of size (maximum element + 1) and initialize all elements to 0.
+3. Count the occurrences of each element in the input array and store it in the count array.
+4. Modify the count array such that each element at each index stores the sum of previous counts.
+5. Output each element from the input array into the correct sorted position in the output array.
+
+## 3. How does Counting Sort work?
+
+- Counting Sort calculates the position of each element in the sorted array by using the count array.
+- The count array stores the count of each unique element.
+- By iterating through the input array and the count array, elements are placed into their correct positions in the output array.
+
+## 4. Problem Description
+
+Given an array of integers, implement the Counting Sort algorithm to sort the array in ascending order.
+
+## 5. Examples
+
+**Example 1:**
+```markdown
+Input: [4, 2, 2, 8, 3, 3, 1]
+Output: [1, 2, 2, 3, 3, 4, 8]
+
+**Example 2:**
+```markdown
+Input: [7, 4, 4, 6, 2, 1, 5, 3, 8, 5]
+Output: [1, 2, 3, 4, 4, 5, 5, 6, 7, 8]
+```
+
+**Explanation of Example 1:**
+- The initial array is [4, 2, 2, 8, 3, 3, 1].
+- Count array: [0, 1, 2, 2, 1, 0, 0, 0, 1]
+- Sorted array: [1, 2, 2, 3, 3, 4, 8]
+
+## 6. Constraints
+
```

Comment:
replace 

```md

- $The array can have any number of elements.$
- $All elements in the array are non-negative integers.$
```

to 

```md

- The array can have any number of elements.
- All elements in the array are non-negative integers.
```

> Since we use `$` for math equations with KaTeX, it's not necessary to use it everywhere.